server/grpc_service.go

```diff
@@ -246,6 +250,25 @@ var errSendRegionHeartbeatTimeout = errors.New("send region heartbeat timeout")
 type heartbeatServer struct {
 	stream pdpb.PD_RegionHeartbeatServer
 	closed int32
+	active int32
+}
+
+func newHeartbeatServer(stream pdpb.PD_RegionHeartbeatServer) *heartbeatServer {
+	s := &heartbeatServer{stream: stream}
+	go func() {
+		for {
+			time.Sleep(regionHeartbeatStreamMaxIdleTime)
```

Comment:
seem we can't quit this goroutine quickly, if the stream is connected and closed frequently, we may have many goroutines.